SU2_CFD/src/output/COutput.cpp

```diff
@@ -347,15 +356,57 @@ void COutput::WriteToFile(CConfig *config, CGeometry *geometry, unsigned short f
 
     case RESTART_BINARY:
 
-      if (fileName.empty())
+      if (fileName.empty()){
         fileName = config->GetFilename(restartFilename, "", curTimeIter);
 
+        if (!config->GetWrt_Restart_Overwrite()){
+          if (config->GetMultizone_Problem()){
+            /* --- do nothing --- */  
+          }
+          else {
+
+            // append previous iteration to filename
+            filename_iter = fileName;
+            // for the final iteration we need to add 1
+            std::stringstream inner_iter_ss;
+            if (config->GetnIter() == curInnerIter+1){ 
+              inner_iter_ss << "_" << std::setw(8) << std::setfill('0') << previous_inner_iteration+1;
+              filename_iter.append((inner_iter_ss).str());
+            }
+            else {   
+              inner_iter_ss << "_" << std::setw(8) << std::setfill('0') << previous_inner_iteration;
+              filename_iter.append(inner_iter_ss.str());
+            }
+            filename_iter.append(CSU2BinaryFileWriter::fileExt);
+
+             /*--- rename previous restart.dat to restart_xxxx.dat ---*/
+            rename(restartFilename.c_str(),filename_iter.c_str()); 
```

Comment:
Should this be done by all ranks or only master?

Reply:
yep, master only.